### PR TITLE
If vm is in a service, use svc uuid in volume name

### PR DESCRIPTION
**Pull request by cjellick**
This allows for reuse of volumes if a vm goes down because it fails
healthchecks.

### CODE REVIEW

File: code/iaas/config-item/server/src/main/java/io/cattle/platform/configitem/context/dao/impl/MetaDataInfoDaoImpl.java [hunk 1]
```diff
@@ -1,18 +1,19 @@
...
-import static io.cattle.platform.core.model.tables.ServiceIndexTable.SERVICE_INDEX;
+import static io.cattle.platform.core.model.tables.HostIpAddressMapTable.*;
+import static io.cattle.platform.core.model.tables.HostTable.*;
+import static io.cattle.platform.core.model.tables.InstanceHostMapTable.*;
+import static io.cattle.platform.core.model.tables.InstanceTable.*;
+import static io.cattle.platform.core.model.tables.IpAddressNicMapTable.*;
+import static io.cattle.platform.core.model.tables.IpAddressTable.*;
+import static io.cattle.platform.core.model.tables.NicTable.*;
+import static io.cattle.platform.core.model.tables.ServiceExposeMapTable.*;
+import static io.cattle.platform.core.model.tables.ServiceIndexTable.*;
```
Comment (cjellick) [inline code/iaas/config-item/server/src/main/java/io/cattle/platform/configitem/context/dao/impl/MetaDataInfoDaoImpl.java]: blame cmd+shift+O

Comment (ibuildthecloud) [inline code/iaas/config-item/server/src/main/java/io/cattle/platform/configitem/context/dao/impl/MetaDataInfoDaoImpl.java]: I like `.*` in this situation.

File: code/implementation/vm/src/main/java/io/cattle/platform/vm/process/VirtualMachinePreCreate.java [hunk 2]
```diff
@@ -98,10 +102,20 @@ public HandlerResult handle(ProcessState state, ProcessInstance process) {
         Object objectDisks = DataAccessor.field(instance, InstanceConstants.FIELD_DISKS, Object.class);
         if (objectDisks instanceof List<?>) {
             String namePrefix = instance.getName();
+            Long svcIndexId = DataAccessor.fieldLong(instance, InstanceConstants.FIELD_SERVICE_INSTANCE_SERVICE_INDEX_ID);
+
+            String uuidPart = null;
+            if (svcIndexId != null) {
+                Service svc = serviceDao.getServiceByServiceIndexId(svcIndexId);
+                uuidPart = svc.getUuid().substring(0, 7);
```
Comment (ibuildthecloud) [inline code/implementation/vm/src/main/java/io/cattle/platform/vm/process/VirtualMachinePreCreate.java]: Can we add the service index to that volume name.  So if I scale the service to more than 1 I don't end up trying to use the same volume.

Comment (ibuildthecloud) [inline code/implementation/vm/src/main/java/io/cattle/platform/vm/process/VirtualMachinePreCreate.java]: Nevermind, the id is already there in the container name.

**Comment by ibuildthecloud** [comment]
LGTM

**Pull request merged**